torch/nn/_functions/packing.py

```diff
@@ -5,6 +5,8 @@
 class PackPadded(Function):
     @staticmethod
     def forward(ctx, input, lengths, batch_first):
+        lengths = list(lengths)
```

Comment:
Is there a reason why we have to do this? It should work just fine with a Variable, right?

Reply:
there are some minor things, like calling reversed(Variable) doesn't work. I'll go through all these and double check if I can simplify them.

Reply:
I think that's the only issue (but of course fast reversal / negative strides are a longstanding gap in TH)

Reply:
Can we replace `reversed` with a bit more verbose iteration over indices backward, and slice the var at every step.